faf-commons-data/src/main/java/com/faforever/commons/replay/header/GameOptions.java

```diff
@@ -0,0 +1,126 @@
+package com.faforever.commons.replay.header;
+
+import java.util.Arrays;
+
+/**
+ * Populated by the game options field of the table that is passed to `CLobby:LaunchGame`
+ * @param autoTeams
+ * @param teamLock
+ * @param teamSpawn
+ * @param allowObservers
+ * @param cheatsEnabled
+ * @param prebuiltUnits
+ * @param revealedCivilians
+ * @param scoreEnabled
+ * @param unitCap
+ * @param unRated
+ * @param victory
+ */
+public record GameOptions(AutoTeams autoTeams, TeamLock teamLock, TeamSpawn teamSpawn, boolean allowObservers,
+                          boolean cheatsEnabled, boolean prebuiltUnits, boolean revealedCivilians, boolean scoreEnabled,
+                          int unitCap, boolean unRated, Victory victory) {
+
+  public enum AutoTeams {
+    NONE("none", "None"),
+    MANUAL("manual", "Manual"),
+    TOP_VS_BOTTOM("tvsb", "Top versus bottom"),
+    LEFT_VS_RIGHT("lvsr", "Left versus right"),
+    EVEN_VS_UNEVEN("pvsi", "Even versus uneven");
+
+    public final String readable;
+
+    private final String key;
+
+    AutoTeams(String key, String readable) {
+      this.readable = readable;
+      this.key = key;
+    }
+
+    public static AutoTeams findByKey(String key) {
+      for (AutoTeams autoTeam : values()) {
+        if (autoTeam.key.equals(key)) {
+          return autoTeam;
+        }
+      }
+      throw new IllegalArgumentException("Unknown key: " + key);
+    }
+  }
+
+  public enum TeamLock {
+    LOCKED("locked", "Locked"), UNLOCKED("unlocked", "Unlocked");
+
+    public final String readable;
+
+    private final String key;
+
+    TeamLock(String key, String readable) {
+      this.key = key;
+      this.readable = readable;
+    }
+
+    public static TeamLock findByKey(String key) {
+      for (TeamLock teamLock : values()) {
+        if (teamLock.key.equals(key)) {
+          return teamLock;
+        }
+      }
+      throw new IllegalArgumentException("Unknown key: " + key);
+    }
+  }
+
+
+  public enum TeamSpawn {
+    fixed("fixed", "Fixed"),
+    random("random", "Random"),
+    balanced("balanced", "Balanced"),
+    balanced_flex("balanced_flex", "Flexible balanced"),
+    random_reveal("random_reveal", "Random and revealed"),
+    balanced_reveal("balanced_reveal", "Balanced and revealed"),
+    balanced_reveal_mirrored("balanced_reveal_mirrored", "Mirror balanced and revealed"),
+    balanced_flex_reveal("balanced_flex_reveal", "Flexible balanced and revealed");
```

Comment:
These enums should be properly cased as well

Reply:
Processed in: https://github.com/FAForever/faf-java-commons/pull/143/commits/bdee610917065cc31e0bd517ad69b7fcfb9dcea6